DatadogCrashReporting/Sources/CrashContext/CrashContext.swift

```diff
@@ -69,6 +69,9 @@ internal struct CrashContext: Codable, Equatable {
     /// The last _"Is app in foreground?"_ information from crashed app process.
     let lastIsAppInForeground: Bool
 
+    /// The override for `source_type` for crashes.
+    let nativeSourceTypeOverride: String?
```

Comment:
I don't think we need to encode this value through `CrashContext` to have it available upon app restart (when we actually send the crash). With setting `nativeSourceOverride` in `DatadogContext` on app init, it will be set to the same value in next app session, right? That assumed, we can read it from `eventWriteContext` API before writing the actual Log event or RUM events:

- for RUM:
https://github.com/DataDog/dd-sdk-ios/blob/bdab1a1aab4b0f3e4c08d10883c508a44e435520/DatadogRUM/Sources/Integrations/CrashReportReceiver.swift#L320-L323

- for Logs:
https://github.com/DataDog/dd-sdk-ios/blob/bdab1a1aab4b0f3e4c08d10883c508a44e435520/DatadogLogs/Sources/Feature/MessageReceivers.swift#L264-L266

In both cases, we get `DatadogContext` as the first argument of the lambda. This is how we can make it simpler, given RUM example:
```swift
core.scope(for: RUMFeature.name)?.eventWriteContext(bypassConsent: true) { context, writer in
    let rumError = createRUMError(from: crashReport, and: updatedRUMView, crashDate: realCrashDate, sourceType: context.nativeSourceOverride)
    writer.write(value: rumError)
    writer.write(value: updatedRUMView)
}
```
I wouldn't mind creating error events on the context thread that this change would imply. This is negligible impact, but we can save a lot of hassle by not coding this simple value through `CrashContext`.

LTMWDYT @fuzzybinary

Reply:
Yeah, this was the question I think I had initially. I wasn't sure if the CrashReporting feature should send it over the message bus or if it should be pulled from context during the write.  If we're good with pulling it during the write that definitely simplifies it.

I'll adjust the PR.

Reply:
Let's do 👍